### PR TITLE
filterprocessor: add record level log filtering

### PR DESCRIPTION
**Pull request by pmalek-sumo**
**Description:** Adding a feature for filterprocessor: add record level log filtering

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5279

**Testing:** Added unit tests

**Documentation:** Added appropriate changes in README.md and exemplar yaml config

### CODE REVIEW

File: processor/filterprocessor/filter_processor_logs.go [hunk 1]
```diff
@@ -70,31 +96,97 @@ func createLogsMatcher(lp *LogMatchProperties) (filtermatcher.AttributesMatcher,
...
+
 // shouldSkipLogsForResource determines if a log should be processed.
 // True is returned when a log should be skipped.
 // False is returned when a log should not be skipped.
 // The logic determining if a log should be skipped is set in the resource attribute configuration.
 func (flp *filterLogProcessor) shouldSkipLogsForResource(resource pdata.Resource) bool {
+	if flp.includeResources == nil && flp.excludeResources == nil {
+		return false
+	}
+
```
Comment (dmitryax) [inline processor/filterprocessor/filter_processor_logs.go:189]: This part looks unrelated to the PR's goal. 

And is there any benefit from this change?

**Review by dmitryax — COMMENTED** [review]
Just one nit, otherwise looks good to me!

File: processor/filterprocessor/filter_processor_logs.go (hunk 1 above)
Comment (pmalek-sumo) [reply processor/filterprocessor/filter_processor_logs.go]: Indeed, it should have ended up here. Thanks! Removed.

**Review by dmitryax — APPROVED** [review]
